src/test/java/org/isf/medicalsinventory/Tests.java

```diff
@@ -572,4 +637,180 @@ void testDeleteInventory() throws Exception {
+		assertThat(medicalInventoryRows).isNotEmpty();
+		assertThat(medicalInventoryRows).hasSize(1);
+		medicalInventoryManager.validateMedicalInventoryRow(savedInventory, medicalInventoryRows);
+		int inventoryId = inventory.getId();
+		String status = InventoryStatus.validated.toString();
+		inventory = medicalInventoryIoOperation.getInventoryById(inventoryId);
+		assertThat(inventory).isNotNull();
+		assertThat(inventory.getStatus()).isEqualTo(status);
+	}
+
+	@ParameterizedTest(name = "Test with AUTOMATICLOT_IN={0}, AUTOMATICLOT_OUT={1}, AUTOMATICLOTWARD_TOWARD={2}")
+	@MethodSource("automaticlot")
+	void testConfirmMedicalInventory(boolean in, boolean out, boolean toward) throws Exception {
+		setGeneralData(in, out, toward);
+		Ward ward = testWard.setup(false);
+		wardIoOperationRepository.saveAndFlush(ward);
+		MovementType chargeType = new MovementType("inventory+", "Inventory+", "+", "non-operational");
+		MovementType dischargeType = new MovementType("inventory-", "Inventory-", "-", "non-operational");
+		Supplier supplier = new Supplier(1, "INVENTORY", null, null, null, null, null, null);
+		Ward destination = new Ward("INV", "ward inventory", null, null, null, 8, 1, 1, false, false);
+		dischargeType = medicalDsrStockMovementTypeIoOperationRepository.save(dischargeType);
+		chargeType = medicalDsrStockMovementTypeIoOperationRepository.save(chargeType);
+		supplier = supplierIoOperationRepository.save(supplier);
+		destination = wardIoOperationRepository.save(destination);
+		MedicalInventory inventory = testMedicalInventory.setup(ward, false);
+		inventory.setChargeType(chargeType.getCode());
+		inventory.setDestination(destination.getCode());
+		inventory.setSupplier(supplier.getSupId());
+		inventory.setDischargeType(dischargeType.getCode());
+		inventory = medicalInventoryIoOperation.newMedicalInventory(inventory);
+		MedicalType medicalType = testMedicalType.setup(false);
+		Medical medical = testMedical.setup(medicalType, false);
+		Lot lotOne = testLot.setup(medical, false);
+		Movement firstMovement = testMovement.setup(medical, chargeType, ward, lotOne, supplier, false);
+		firstMovement.setQuantity(100);
+		MedicalStock firstmedicalStock = testMedicalStock.setup(firstMovement);
+		Lot lotTwo = testLot.setup(medical, false);
+		lotTwo.setCode("LOT-001");
+		Movement secondMovement = testMovement.setup(medical, chargeType, ward, lotTwo, supplier, false);
+		secondMovement.setQuantity(100);
+		MedicalStock secondmedicalStock = testMedicalStock.setup(firstMovement);
+		Lot lotThree = testLot.setup(medical, false);
+		lotTwo.setCode("LOT-003");
+		medicalType = medicalTypeIoOperationRepository.save(medicalType);
+		medical = medicalsIoOperationRepository.save(medical);
+		lotOne = lotIoOperationRepository.save(lotOne);
+		lotTwo = lotIoOperationRepository.save(lotTwo);
+		lotThree = lotIoOperationRepository.save(lotThree);
+		firstMovement = medicalStockIoOperation.newMovement(firstMovement);
```

Comment:
`firstMovement` is not used in this method after this assignment; can be removed